script.js

```diff
@@ -0,0 +1,74 @@
+let lengthInput = document.getElementById("length");
+let outputContainer = document.getElementById("output-container");
+let rotateButtonContainer = document.getElementById("rotateButtonContainer");
+let rotatedContainer = document.getElementById("rotated-container");
+function createTriangle(length) {
+    let result = "";
+    for (let i = 1; i <= length; i++) {
+        for (let j = length; j > i; j--) {
+            result += "";
+        }
+        for (let j = 1; j <= i; j++) {
+            result += "*" + " ";
+        }
+        result += "\n";
+    }
+    return result;
+}
+
+function rotateTriangle(triangle) {
+    const lines = triangle.trim().split("\n");
+    const length = lines.length;
+    let rotatedTriangle = "";
+    for (let i = 0; i < length; i++) {
+        rotatedTriangle += "".repeat(i);
+        rotatedTriangle += lines[length - i - 1];
+        rotatedTriangle += "\n";
+    }
+    return rotatedTriangle;
+}
+
+
+function resetPage() {
+    lengthInput.value = "";
+    outputContainer.innerHTML = "";
+    rotateButtonContainer.style.display = "none";
+    rotatedContainer.innerHTML = "";
+    lengthInput.disabled = false;
+    lengthInput.focus();
+}
+
+
+function handleKeyPress(event) {
+    if (event.key === "Enter") {
+        const length = parseInt(lengthInput.value);
+        if (isNaN(length) || length >= 25) {
+            alert("Invalid input! Please enter a number less than 25.");
+            resetPage();
+        } else {
+            const triangle = createTriangle(length);
+            outputContainer.innerHTML = "<pre>" + triangle + "</pre>";
+            rotateButtonContainer.style.display = "block";
+            document.getElementById("rotateButton").disabled = false;
```

Comment:
this type of stuff, when you traversing DOM to find element is expensive operation, so you better cache it.
something like  
const rotateButton = document.getElementById("rotateButton");
then whereever you need it, just use reference rotateButton

Reply:
Okay, will change it. Thanks